tests/lib/nested.sh

```diff
@@ -265,111 +265,188 @@ cleanup_nested_env(){
             ;;
         *)
             echo "unsupported system"
             exit 1
             ;;
         esac
+}
 
-        if [ "$BUILD_SNAPD_FROM_CURRENT" = "true" ]; then
-            if is_core_16_nested_system; then
-                repack_snapd_deb_into_core_snap "$WORK_DIR"
-                EXTRA_FUNDAMENTAL="$EXTRA_FUNDAMENTAL --snap $WORK_DIR/core-from-snapd-deb.snap"
-
-            elif is_core_18_nested_system; then
-                repack_snapd_deb_into_snapd_snap "$WORK_DIR"
-                EXTRA_FUNDAMENTAL="$EXTRA_FUNDAMENTAL --snap $WORK_DIR/snapd-from-deb.snap"
-
-            elif is_core_20_nested_system; then
-                snap download --basename=pc-kernel --channel="20/edge" pc-kernel
-                uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$WORK_DIR/image"
-
-                # Get the snakeoil key and cert
-                KEY_NAME=$(get_snakeoil_key)
-                SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
-                SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
-
-                # Prepare the pc kernel snap
-                KERNEL_SNAP=$(ls "$WORK_DIR"/image/pc-kernel_*.snap)
-                KERNEL_UNPACKED="$WORK_DIR"/image/kernel-unpacked
-                unsquashfs -d "$KERNEL_UNPACKED" "$KERNEL_SNAP"
-                sbattach --remove "$KERNEL_UNPACKED/kernel.efi"
-                sbsign --key "$SNAKEOIL_KEY" --cert "$SNAKEOIL_CERT" "$KERNEL_UNPACKED/kernel.efi"  --output "$KERNEL_UNPACKED/kernel.efi"
-                snap pack "$KERNEL_UNPACKED" "$WORK_DIR/image"
-
-                chmod 0600 "$KERNEL_SNAP"
-                rm -f "$PWD/pc-kernel.snap"
-                rm -rf "$KERNEL_UNPACKED"
-                EXTRA_FUNDAMENTAL="--snap $KERNEL_SNAP"
-
-                # Prepare the pc gadget snap (unless provided by extra-snaps)
-                GADGET_SNAP=""
-                if [ -d extra-snaps ]; then
-                    GADGET_SNAP=$(find extra-snaps -name 'pc_*.snap')
-                fi
-                # XXX: deal with [ "$ENABLE_SECURE_BOOT" != "true" ] && [ "$ENABLE_TPM" != "true" ]
-                if [ -z "$GADGET_SNAP" ]; then
-                    snap download --basename=pc --channel="20/edge" pc
-                    unsquashfs -d pc-gadget pc.snap
-                    secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
-                    snap pack pc-gadget/ "$WORK_DIR/image"
-
-                    GADGET_SNAP=$(ls "$WORK_DIR"/image/pc_*.snap)
-                    rm -f "$PWD/pc.snap" "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
-                    EXTRA_FUNDAMENTAL="$EXTRA_FUNDAMENTAL --snap $GADGET_SNAP"
+create_nested_core_vm(){
+    # shellcheck source=tests/lib/prepare.sh
+    . "$TESTSLIB"/prepare.sh
+    # shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+
+    local IMAGE_DIR IMAGE_NAME
+    IMAGE_DIR="$(get_image_dir)"
+    IMAGE_NAME="$(get_image_name core)"
+
+    prepare_image_dir
+    if [ ! -f "$IMAGE_DIR/$IMAGE_NAME" ]; then
```

Comment:
if the image name exists then this function just silently exits, correct?

Reply:
yes, in the following pr, we will check here if the image was previously built, so it can be reused